decidim-api/app/controllers/decidim/api/sessions_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Api
+    class SessionsController < ::Devise::SessionsController
+      skip_before_action :verify_authenticity_token
+
+      respond_to :json
+
+      def failure
+        render json: anonymous_user.serializable_hash.slice("id", "name", "nickname").merge(
```

Comment:
This should not be repeated. Also, you can use `user.attributes.slice(...)`.

Reply:
*anonymous_user